pilot/pkg/xds/ads.go

```diff
@@ -636,6 +636,10 @@ func (s *DiscoveryServer) computeProxyState(proxy *model.Proxy, request *model.P
 			case gvk.Ingress:
 				sidecar = true
 				gateway = true
+			case gvk.ReferencePolicy:
+				if rp := s.Env.ConfigStore.Get(gvk.ReferencePolicy, conf.Name, conf.Namespace); isSecretReferencePolicy(rp) {
+					gateway = true
+				}
```

Comment:
```suggestion
					gateway = true
```

I would just keep it simple here like the rest. This computation is not *that* expensive that we need to be this precise